src/module/rules/rule-element/grant-item/rule-element.ts

```diff
@@ -321,12 +347,11 @@ class GrantItemRuleElement extends RuleElementPF2e<GrantItemSchema> {
                 (setHasElement(PHYSICAL_ITEM_TYPES, grantee.type) ? "detach" : "cascade"),
         };
 
-        if (grantee instanceof ItemPF2e) {
+        grantee.flags = fu.mergeObject(grantee.flags ?? {}, { pf2e: { grantedBy } });
+        if (grantee instanceof ItemPF2e && grantee._id) {
             // This is a previously granted item: update its grantedBy flag
             // Don't await since it will trigger a data reset, possibly wiping temporary roll options
```

Comment:
Why should this comment remain if there's nothing awaitable?

Reply:
It should not, I failed to catch it in my victory lab when I was trying to find out what parts we actually needed.